### PR TITLE
CESQL

### PR DESCRIPTION
**Pull request by slinkydeveloper**
This PR provides a new package that implements CESQL spec.

The package passes all the tests of the TCK. This implementation is not super efficient, but I think it's good as a first step.

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### CODE REVIEW

**Comment by n3wscott** [comment]
What is the code coverage look like on the new code?

**Comment by slinkydeveloper** [comment]
> What is the code coverage look like on the new code?

TCK covers around 80%, most non covered paths are the error paths (which just shortcircuit the computation)

**Comment by n3wscott** [comment]
I wonder if nats changed something... or it needs more time... that is not related to your PR...

**Comment by n3wscott** [comment]
I am assuming a readme will come in a followup, one nit is if you want that module to be released next push, please add the package path to the correct places in the `./hack/tag-release.sh` script, but you can do this in a followup as well.

LGTM

**Comment by slinkydeveloper** [comment]
> I wonder if nats changed something... or it needs more time... that is not related to your PR...

There is an open issue https://github.com/cloudevents/sdk-go/issues/630

> I am assuming a readme will come in a followup, one nit is if you want that module to be released next push, please add the package path to the correct places in the ./hack/tag-release.sh script, but you can do this in a followup as well.

Both fixed, can you check it? This should be ready for the first release, i also made very explicit that the package APIs are subject to changes.

**Review by n3wscott — APPROVED** [review]
LGTM

**Comment by n3wscott** [comment]
you can follow the manual release process to produce a `v2.4.1` release of the sql module, cherry-pick this commit to the `release-2.4` branch.

**Comment by n3wscott** [comment]
I landed the headers change, you can run the tool on the sql dir like: `./hack/boilerplate/add-boilerplate.sh go sql`

**Comment by slinkydeveloper** [comment]
Coyright headers done.

> you can follow the manual release process to produce a v2.4.1 release of the sql module, cherry-pick this commit to the release-2.4 branch.

No need to have a release now about it. We can wait 2.5, the spec is still in the works.

File: sql/v2/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+## User guide
+
+To start using it:
+
+```go
+import cesqlparser "github.com/cloudevents/sdk-go/sql/v2/parser"
+
+// Parse the expression
+expression, err := cesqlparser.Parse("subject = 'Hello world'")
```
Comment (matzew) [inline sql/v2/README.md:15]: mind adding a bit more of use cases, also more complex ones?

Sure: can be done in a follow up PR

**Comment by n3wscott** [comment]
I would favor landing what you have and continuing to make PRs on it. We can choose to release it or not when it is time for the `v2.5.0`

LGTM

**Pull request merged**